### ISSUE TITLE
Open file from sftp by URI (with new info)

### ISSUE THREAD

**Issue by C1977** [satiromarra/vscode-sftp#75]
I suggest a feature to open a file from sftp direct by calling vscode with URL parameter like described here:
[https://www.eliostruyf.com/callback-extension-vscode/]

How to read this parameters is described in the link or at command registerUriHandler

Would be very helpful to be able to open VSCode and directly open a file to edit. I made some tests with my extension to read the parameters and it worked. Maybe we can set the connectionid and the path by this Get-Parameters.

**Comment by satiromarra** [comment]
@C1977 Thanks for the info. I have done some tests and it works. However, I still have problems when displaying the file in "remote explorer", It's a problem that came from before.

Something that has to be done and that is not in the blog, is to add in `package.json` in `activationEvents` a value `onUri`
@cfunke2 For your interest.

**Comment by C1977** [comment]
Its me, C1977 = cfunke2

**Comment by cfunke2** [comment]
Hello Satiro, did you do any further trys? Maybe I can help you in any way, I made a plugin for my company (just to avoid that two people work on same file). Would be so cool if we could open VSCode with a specific file and line number by URL

**Comment by satiromarra** [comment]
I have done some tests but I have not been completely satisfied.
Remotely you can't work and it would be necessary to download the file before editing it so:
1.- Open vscode with the path to the file.
2.- Download the file (overwrite the local file), This requires some validations so as not to lose local data.
3.- Open the file to edit.

**Comment by cfunke2** [comment]
Yes this sounds very good. For 2. - I think it would be okay to just show an error "File already opened" in that case. The user can close it and just click the link again if we want

**Comment by C1977** [comment]
Any change to make progress? Can I contact you by eMail?

**Comment by satiromarra** [comment]
Hi @C1977 

Lately I've been saturated with work and I can't give it the time I need.

As soon as I can, I'm going to continue with it.

**Comment by C1977** [comment]
We would also donate something to it if that helps.

**Comment by C1977** [comment]
If there is anything I could do to help you move forward at this point, please let me know. That would be a super big help for my team and it would be worth a lot to us. Best regards from Eime, Germany

**Comment by C1977** [comment]
We build this ourselv now by an additional plugin.
